datasets/flwr_datasets/utils.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utils for FederatedDatasets."""
+
+
+from typing import Dict
+
+from flwr_datasets.partitioner import IidPartitioner, Partitioner
+
+
+def _instantiate_partitioners(partitioners: Dict[str, int]) -> Dict[str, Partitioner]:
+    """Transform the partitioners from the initial format to instantiated objects.
+
+    Parameters
+    ----------
+    partitioners: Dict[str, int]
+        Partitioners specified as split to the number of partitions format.
+
+    Returns
+    -------
+    partitioners: Dict[str, Partitioner]
+        Partitioners specified as split to Partitioner object.
+    """
+    instantiated_partitioners: Dict[str, Partitioner] = {}
+    for key, value in partitioners.items():
+        instantiated_partitioners[key] = IidPartitioner(num_partitions=value)
+    return instantiated_partitioners
+
+
+def _check_if_dataset_supported(dataset: str) -> None:
+    """Check if the dataset is in the narrowed down list of the tested datasets."""
+    supported_datasets = [
+        "mnist",
+        "cifar10",
+        "fashion_mnist",
+        "food101",
+        "sasha/dog-food",
+        "zh-plus/tiny-imagenet",
```

Comment:
Did we test these thoroughly?

Reply:
@danieljanes I can add them easily to the test bc I created parametrized tests, but I would discourage it because it'll take a lot of time to download them.
But to answer the question, I manually checked them. They work because:
* they have columns that are named exactly the same ("image", "label")
* they are also image datasets
Two of them don't have the test split but have a validation split. But that's up to a user to use an existent split. 
Let me know what you think.

Reply:
Let's remove `food101` (>5gb), and have one test that also uses `fashion_mnist`, `tiny-imagenet`, and `dog-food` (they are all below 300mb, and we can probably cache them on the CI)

Reply:
Done